src/textual/await_complete.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from asyncio import Future, gather, wait
+from typing import Any, Coroutine, Generator, Generic, TypeVar
+
+ReturnType = TypeVar("ReturnType")
+
+
+class AwaitComplete(Generic[ReturnType]):
+    """An 'optionally-awaitable' object."""
+
+    _instances: set["AwaitComplete"] = set()
```

Comment:
This makes me a little nervous in they are at the class level. If you have multiple apps, then they could clash.

I know its an exotic scenario at the moment, but I can see a use for apps in apps.

Do we need this at all? I see its used in `_wait_all`, but I don't see where that is used. Unless GitHub has hidden it for me.